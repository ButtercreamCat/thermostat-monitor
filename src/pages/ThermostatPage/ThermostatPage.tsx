import React, { Component } from "react";
import "./ThermostatPage.scss";
import Thermostat from "../../components/Thermostat/Thermostat";
import StatusButtons from "../../components/StatusButtons/StatusButtons";

type ThermostatPageState = {
  status: string;
};

export default class ThermostatPage extends Component<{}, ThermostatPageState> {
  state = {
    status: "Off"
  };

  handleClick = (status: string) => {
    this.setState({
      status
    });
  };

  render() {
    return (
      <section className="unitInfo_content">
        <Thermostat status={this.state.status}></Thermostat>
        <StatusButtons
          handleClick={this.handleClick}
          status={this.state.status}
        ></StatusButtons>
      </section>
    );
  }
}
