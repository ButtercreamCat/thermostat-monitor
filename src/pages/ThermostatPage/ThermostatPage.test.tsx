import React from "react";
import { shallow } from "enzyme";
import ThermostatPage from "./ThermostatPage";

test("should render ThermostatPage correctly", () => {
  const wrapper = shallow(<ThermostatPage></ThermostatPage>);
  expect(wrapper).toMatchSnapshot();
});

test("should set state correctly on status button click", () => {
  const wrapper = shallow(<ThermostatPage></ThermostatPage>);
  const handleClickProp: (status: string) => void = wrapper
    .find("StatusButtons")
    .prop("handleClick");
  handleClickProp("Auto");
  expect(wrapper.state("status")).toBe("Auto");
});
