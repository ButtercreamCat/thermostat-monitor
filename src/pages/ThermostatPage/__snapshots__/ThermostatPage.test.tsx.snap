// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ThermostatPage correctly 1`] = `
<section
  className="unitInfo_content"
>
  <Thermostat
    status="Off"
  />
  <StatusButtons
    handleClick={[Function]}
    status="Off"
  />
</section>
`;
